container-host-files/opt/scf/bin/klog.sh

```diff
@@ -5,192 +5,94 @@ set -e
 KLOG=${HOME}/klog
 
 if [ "$1" == "-h" ]; then
-    cat <<EOF
+  cat <<EOF
 usage: $0 [-f] [-v] [INSTANCE_ID]
 
   -f  forces fetching of all logs even if a cache already exists
 
   INSTANCE_ID defaults to "scf"
 EOF
-    exit
+  exit
 fi
 
 FORCE=0
 if [ "$1" == "-f" ]; then
-    shift
-    FORCE=1
+  shift
+  FORCE=1
 fi
 
 NS=${1-scf}
 DONE="${KLOG}/${NS}/done"
 
 if [ "${FORCE}" == "1" ] ; then
-    rm -f "${DONE}" 2> /dev/null
+  rm -f "${DONE}" 2> /dev/null
 fi
 
 function get_phase() {
   kubectl get pod "${POD}" --namespace "${NS}" --output=jsonpath='{.status.phase}'
 }
 
 function check_for_log_dir() {
-  kubectl exec "${POD}" --namespace "${NS}" -- bash -c "[ -d /var/vcap/sys/log ]" 2> /dev/null
+  kubectl exec "${POD}" --namespace "${NS}" --container "${CONTAINER}" -- bash -c "[ -d /var/vcap/sys/log/"${CONTAINER}" ]" 2> /dev/null
 }
 
 if [ ! -f "${DONE}" ]; then
-    rm -rf "${KLOG:?}/${NS:?}"
-    mkdir -p "${KLOG}/${NS}"
+  rm -rf "${KLOG:?}/${NS:?}"
+  BASE_DIR="${KLOG}/${NS}"
 
-    PODS=($(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\?\///'))
+  # Retrieving all the pods
+  PODS=($(kubectl get pods --namespace "${NS}" --output=jsonpath='{.items[*].metadata.name}'))
 
-    for POD in "${PODS[@]}"; do
-        DIR="${KLOG}/${NS}/${POD}"
+  for POD in "${PODS[@]}"; do
+    POD_DIR="${BASE_DIR}/${POD}"
 
-        mkdir -p "${DIR}"
+    # Retrieving all the containers within a pod
+    CONTAINERS=($(kubectl get pods "${POD}" --namespace "${NS}" --output=jsonpath='{.spec.containers[*].name}'))
 
-        # Get the CF logs inside the pod if there are any
-        if [ "$(get_phase)" != 'Succeeded' ] && check_for_log_dir; then
-            kubectl cp --namespace "${NS}" "${POD}":/var/vcap/sys/log/ "${DIR}/" 2> /dev/null
-        fi
+    # Iterate over containers and dump logs
+    for CONTAINER in "${CONTAINERS[@]}"; do
 
-        # Get the pod logs - previous may not be there if it was successful on the first run.
-        # Unfortunately we can't get anything past the previous one
-        kubectl logs "${POD}" --namespace "${NS}" > "${DIR}/kube.log"
-        kubectl logs "${POD}" --namespace "${NS}" --previous > "${DIR}/kube-previous.log" 2> /dev/null || true
-        kubectl describe pods "${POD}" --namespace "${NS}" > "${DIR}/describe-pod.txt"
+      CON_DIR="${POD_DIR}/${CONTAINER}"
+      mkdir -p ${CON_DIR}
+
+      # Get the CF logs inside the pod if there are any
```

Comment:
[nit] Since you are touching this file, could you add missing punctuation to the comments? (here and elsewhere)

Reply:
Check and see anything else to add would be useful.